sources/packages/backend/apps/api/src/route-controllers/student-account-applications/_test_/e2e/student-account-application.aest.controller.approveStudentAccountApplication.e2e-spec.ts

```diff
@@ -0,0 +1,155 @@
+      AppAESTModule,
+      FormService,
+    );
+
+    // Insert a fake email contact to send ministry email.
+    await db.notificationMessage.update(
+      { id: NotificationMessageType.PartialStudentMatchNotification },
+      { emailContacts: ["dummy@some.domain"] },
+    );
+    await db.notification.clear();
+  });
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    // Ensure the SIN used for this method will not conflict.
+    await db.sfasIndividual.update({ sin: "534012702" }, { sin: "000000000" });
+    await db.sfasIndividual.update({ sin: "534012703" }, { sin: "000000000" });
+    await db.sinValidation.update({ sin: "534012702" }, { sin: "000000000" });
+    await db.sinValidation.update({ sin: "534012703" }, { sin: "000000000" });
+  });
+
+  it("Should send a notification message when at least a partial match is found for importing a student record from SFAS.", async () => {
+    // Arrange
+    const user = await db.user.save(createFakeUser());
+    // Submitted data to simulated the exists student account already saved on DB.
+    // Same data will be used to be submitted when Ministry is approving it.
+    const submittedData = {
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      dateOfBirth: "2001-01-31",
+      phone: "1234567890",
+      sinNumber: "534012702",
+      mode: "aest-account-approval",
+      identityProvider: "bceidboth",
+      sinConsent: true,
+      gender: "X",
+      addressLine1: "address 1",
+      city: "Victoria",
+      country: "Canada",
+      postalCode: "H1H1H1H",
+      provinceState: "BC",
+      selectedCountry: "canada",
+      canadaPostalCode: "H1H1H1H",
+    };
+    // Save the fake student account application to be later approved by the Ministry
+    // and create the Student Account.
+    const studentAccountApplication = await db.studentAccountApplication.save(
+      createFakeStudentAccountApplication(
+        { user },
+        { initialValues: { submittedData } },
+      ),
+    );
+
+    await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: user.lastName,
+        birthDate: "2001-01-31",
+        sin: "534012703",
+      },
+    });
+
+    const endpoint = `/aest/student-account-application/${studentAccountApplication.id}/approve`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+    // Mock the form.io response.
+    sharedFormService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: FormNames.StudentProfile,
+      data: { data: submittedData },
+    });
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(submittedData)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+      });
+    // Check that the notification is in the database.
+    const notificationMessageType =
```

Comment:
Minor, the `NotificationMessageType.PartialStudentMatchNotification` is already a "const" and assigning it to a different variable would be recommended only if it improves the code readability, not the IMO.

Reply:
updated